core/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -273,11 +277,73 @@ OpenTelemetryRum build(ServiceManager serviceManager) {
         InitializationEvents initializationEvents = InitializationEvents.get();
         applyConfiguration(serviceManager, initializationEvents);
 
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
Use `android.os.AsyncTask.THREAD_POOL_EXECUTOR` instead (Async task is deprecated but the thread pool still backs almost everything that uses threads on Android). That way you won't spin up a new thread here.

Reply:
Everywhere else in this code bases uses `java.util.concurrent`. Thats why I chose this one over other options. Do you think we should be introducing this to their code base instead of following what they already do?